#VRML_SIM R2023b utf8
# Describe the functionality of your PROTO here.


PROTO SnowBlowerRobot [
  field SFVec3f translation 0 0 0
  field SFRotation rotation 0 0 1 0
  field SFString name "Kevin"
  field SFString controller "<generic>"
]

{
  Robot {
  translation IS translation
  rotation IS rotation
  children [
    Shape {
      appearance DEF BODY_APPEARANCE PBRAppearance {
        baseColor 0.8 1 1
      }
      geometry DEF BODY_GEOMETRY Box {
        size 0.4572 0.3302 0.2794
      }
    }
    Solid {
      translation 0.24 -0.26 -0.15
      children [
        DEF SHROUD_POSE Pose {
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance USE BODY_APPEARANCE
              geometry Mesh {
                url [
                  "../cad/shroud.STL"
                ]
              }
            }
          ]
        }
      ]
      name "shroud"
      boundingObject USE SHROUD_POSE
      physics Physics {
        mass 0.001
      }
    }
    Track {
      translation 0 -0.22 -0.08
      children [
        DEF TRACK_BACK_WHEEL TrackWheel {
          position -0.1524 0
          rotation 0.9088089781155867 0.2950132216685874 -0.2950143053094617 1.666274953948608
          radius 0.0712
        }
        DEF TRACK_FRONT_WHEEL TrackWheel {
          position 0.1524 0
          rotation 0.9088089781155867 0.2950132216685874 -0.2950143053094617 1.666274953948608
          radius 0.0712
        }
        Pose {
          translation -0.1524 0 0
          rotation 1 0 0 1.5708
          children [
            DEF TRACK_WHEEL_SHAPE Shape {
              appearance PBRAppearance {
              }
              geometry DEF TRACK_WHEEL_GEOMETRY Cylinder {
                height 0.1016
                radius 0.0662
              }
            }
          ]
        }
        Pose {
          translation 0.1524 0 0
          rotation 1 0 0 1.5708
          children [
            USE TRACK_WHEEL_SHAPE
          ]
        }
      ]
      name "StbdTrack"
      boundingObject DEF TRACK_BOUNDING_OBJECT Group {
        children [
          Pose {
            translation -0.1524 0 0
            rotation 1 0 0 1.5708
            children [
              DEF TRACK_SHAPE Cylinder {
                height 0.1
                radius 0.0762
              }
            ]
          }
          Pose {
            translation 0.1524 0 0
            rotation 1 0 0 1.5708
            children [
              USE TRACK_SHAPE
            ]
          }
        ]
      }
      physics Physics {
      }
      device [
        LinearMotor {
          name "StbdMotor"
        }
      ]
      animatedGeometry DEF TRACK_ANIMATION Shape {
        appearance PBRAppearance {
          baseColor 0.1 0.1 0.1
          roughness 1
          metalness 0
        }
        geometry DEF TRACK_ANIMATION_GEOMETRY Box {
          size 0.05 0.1016 0.01
        }
      }
      geometriesCount 20
    }
    Track {
      translation 0 0.22 -0.08
      children [
        TrackWheel {
          position -0.1524 0
          rotation 0.0016451401614879355 -0.7071045256247739 0.7071071229699306 3.1383023883109376
          radius 0.0712
        }
        TrackWheel {
          position 0.1524 0
          rotation 0.0016451401614879355 -0.7071045256247739 0.7071071229699306 3.1383023883109376
          radius 0.0712
        }
        Pose {
          translation 0.1524 0 0
          rotation 1 0 0 1.5708
          children [
            USE TRACK_WHEEL_SHAPE
          ]
        }
        Pose {
          translation -0.1524 0 0
          rotation 1 0 0 1.5708
          children [
            USE TRACK_WHEEL_SHAPE
          ]
        }
      ]
      name "PortTrack"
      boundingObject USE TRACK_BOUNDING_OBJECT
      physics Physics {
      }
      device [
        LinearMotor {
          name "PortMotor"
        }
      ]
      animatedGeometry USE TRACK_ANIMATION
      geometriesCount 20
    }
    D435RGB {
      translation 0.53 0 0.17
    }
  ]
  name IS name
  boundingObject USE BODY_GEOMETRY
  physics Physics {
    mass 10
  }
  controller IS controller
  }
}
